src/public-lib/components/checkedSelect/CheckedSelect.tsx

```diff
@@ -86,6 +90,19 @@ export default class CheckedSelect extends React.Component<CheckedSelectProps, {
     {
         return getOptionLabel(option, this.selectedValues, this.props.checkBoxType);
     }
+    
+    @computed get inputValue() {
+        return this.props.inputValue || this.defaultInputValue;
+    }
+    
+    @autobind
+    @action defaultOnInputChange(input:string, options:{action:string}) {
+        // to prevent clearing input value when selecting an option
+        if (options.action !== "set-value") {
+            if (this.props.onInputChange) this.props.onInputChange(input);
+            if (this.props.inputValue === undefined) this.defaultInputValue = input; 
```

Comment:
I think you should set `this.defaultInputValue` regardless, because choosing this.props.inputValue is done in `@computed get inputValue`